ngraph/core/reference/src/runtime/reference/if.cpp

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2018-2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "ngraph/runtime/reference/if.hpp"
+#include "ngraph/runtime/reference/function.hpp"
+namespace ngraph
+{
+    namespace runtime
+    {
+        namespace reference
+        {
+            enum if_body_indexes
+            {
+                then_body_index = 0,
+                else_body_index = 1
+            };
+            void if_reference(
+                const std::vector<std::shared_ptr<Function>>& bodies,
+                const std::vector<op::util::MultiSubgraphOutputDescriptionVector>& out_descs,
+                const std::vector<op::util::MultiSubgraphInputDescriptionVector>& input_descs,
+                const HostTensorVector& out,
+                const HostTensorVector& args)
+            {
+                auto condition_value = args[0]->get_data_ptr<bool>()[0];
```

Comment:
I think we need to check a number of arguments passed:
`NGRAPH_CHECK(args.size() > 0, "If operation must have input condition value.");`
After that we need to check condition type using `get_element_type` and shape.

Reply:
Checking on get_element_type  is bad idea, bool is casting to another type on plugin as result fail on this checking

Reply:
NGRAPH_CHECK(args.size() > 0, "If operation must have input condition value."); was added